Build: Fails with GCC ARM Embedded toolchain v4.9

The mbed-os documentation states that [GCC ARM Embedded toolchain v4.9 should be used](https://docs.mbed.com/docs/mbed-os-handbook/en/5.2/dev_tools/cli/#toolchain-selection):
> If you want to use the GCC ARM Embedded toolchain, set GCC_ARM_PATH to the binary directory of your GCC ARM installation (example: C:\Program Files\GNU Tools ARM Embedded\4.9 2015q2\bin). Use versions 4.9 of GCC ARM Embedded; version 5.0 or any version above may be incompatible with the tools.

Using `gcc-arm-none-eabi-4_9-2015q3-20150921`:
```
$ ./arm-none-eabi-gcc --version
arm-none-eabi-gcc (GNU Tools for ARM Embedded Processors) 4.9.3 20150529 (release) [ARM/embedded-4_9-branch revision 227977]
Copyright (C) 2014 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Building `uvisor` with the recommended toolchain fails with:
```
arm-none-eabi-gcc  -mcpu=cortex-m4 -march=armv7e-m -mthumb  -g3 -Wall -Werror -DUVISOR_PRESENT=1 -DARCH_MPU_ARMv7M -DCONFIGURATION_NUC472_CORTEX_M4_0x20000000_0x0 -DPROGRAM_VERSION=\"0.27.0\"  -I. -I./core  -I./core/cmsis/inc  -I./core/debug/inc  -I./core/lib/printf/inc  -I./core/system/inc  -I./core/vmpu/inc  -I./platform/nuc472/inc -ffunction-sections -fdata-sections -c ./core/system/src/unvic.c -o platform/nuc472/debug/configuration_nuc472_cortex_m4_0x20000000_0x0/./core/unvic.o
./core/system/src/unvic.c: In function 'unvic_init':
./core/system/src/unvic.c:552:5: error: 'for' loop initial declarations are only allowed in C99 or C11 mode
     for (uint32_t ii = 0; ii < NVIC_VECTORS; ii++) {
     ^
./core/system/src/unvic.c:552:5: note: use option -std=c99, -std=gnu99, -std=c11 or -std=gnu11 to compile your code
```

When using a newer version, in my case using Arch Linux's `arm-none-eabi-gcc 6.3.0-1` package, the build succeeds:
```
$ arm-none-eabi-gcc --version
arm-none-eabi-gcc (Arch Repository) 6.3.0
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

--- a/README.md
+++ b/README.md
@@ -41,7 +41,7 @@ To use uVisor on a specific OS, you must complete the porting process for that O
 
 - mbed OS: [NXP FRDM-K64F](http://developer.mbed.org/platforms/FRDM-K64F/).
 
-The Launchpad [GNU ARM Embedded](https://launchpad.net/gcc-arm-embedded) Toolchain builds the uVisor prelinked binary images. Currently, uVisor only supports applications built with this toolchain.
+The Launchpad [GNU ARM Embedded](https://launchpad.net/gcc-arm-embedded) Toolchain builds the uVisor prelinked binary images. Currently, uVisor only supports applications built with this toolchain. The minimum required version is 5.4-2016q3.
 
 ## The uVisor design philosophy
 

